addons/web_editor/controllers/main.py

```diff
@@ -301,130 +301,149 @@ def get_image_info(self, image_id=None, xml_id=None):
+                        continue
+                    url = file_info["url"]
+
+                    # Exclude library files (see regex above)
+                    if excluded_url_matcher.match(url):
+                        continue
+
+                    # Check if the file is customized and get bundle/path info
+                    file_data = self._match_scss_or_js_file_url(url)
+                    if not file_data:
+                        continue
+
+                    # Save info according to the filter (arch will be fetched later)
+                    url_infos[url] = file_data
+
+                    if '/user_custom_' in url \
+                            or file_data['customized'] \
+                            or file_type == 'scss' and not only_user_custom_files:
+                        files_data.append(url)
+
+                    # File data is returned sorted by bundle, with the bundles names and xmlids
+                    if len(files_data):
+                        files_data_by_bundle.append([dict(xmlid=asset_name, name=request.env.ref(asset_name).name), files_data])
+
+        # Filter bundles/files:
+        # - A file which appears in multiple bundles only appears in the first one (the first in the DOM)
+        # - Only keep bundles with files which appears in the asked bundles and only keep those files
+        for i in range(0, len(files_data_by_bundle)):
+            bundle_1 = files_data_by_bundle[i]
+            for j in range(0, len(files_data_by_bundle)):
+                bundle_2 = files_data_by_bundle[j]
+                # In unwanted bundles, keep only the files which are in wanted bundles too (_assets_helpers)
+                if bundle_1[0]["xmlid"] not in bundles_restriction and bundle_2[0]["xmlid"] in bundles_restriction:
+                    bundle_1[1] = [item_1 for item_1 in bundle_1[1] if item_1 in bundle_2[1]]
+        for i in range(0, len(files_data_by_bundle)):
+            bundle_1 = files_data_by_bundle[i]
+            for j in range(i + 1, len(files_data_by_bundle)):
+                bundle_2 = files_data_by_bundle[j]
+                # In every bundle, keep only the files which were not found in previous bundles
+                bundle_2[1] = [item_2 for item_2 in bundle_2[1] if item_2 not in bundle_1[1]]
+
+        # Only keep bundles which still have files and that were requested
+        files_data_by_bundle = [
+            data for data in files_data_by_bundle
+            if (len(data[1]) > 0 and (not bundles_restriction or data[0]["xmlid"] in bundles_restriction))
+        ]
+
+        # Fetch the arch of each kept file, in each bundle
+        for bundle_data in files_data_by_bundle:
+            for i in range(0, len(bundle_data[1])):
+                url = bundle_data[1][i]
+                url_info = url_infos[url]
+
+                content = None
+                if url_info["customized"] and custom_attachments is not None:
```

Comment:
I wonder why that added condition would be useful. If `url_info["customized"]` is true, `custom_attachments` cannot be `None`. Do you know why you added this ?

Reply:
I don't remember why I did this but indeed that makes sense

Reply:
Ok, then remove the added condition (be careful to take my work before push forcing the branch)
Also check pna's review (https://drive.google.com/file/d/1kjYZbTr50GoNijopvMrU7yvqRrF2vX4D/view)

Tell me when you are done so that we can merge this :)

Reply:
@kea14 Finished..?

Reply:
@qsm-odoo  I've just switched on that 15-20mins ago. Still trying to understand the issue with scrolling.